syft/cataloger/cpe.go

```diff
@@ -68,14 +68,43 @@ func candidateTargetSoftwareAttrs(p pkg.Package) []string {
 }
 
 func candidateVendors(p pkg.Package) []string {
-	// TODO: expand with package metadata (from type assert)
-	vendors := []string{p.Name}
-	if p.Language == pkg.Python {
+	vendors := candidateProducts(p)
+	switch p.Language {
+	case pkg.Python:
 		vendors = append(vendors, fmt.Sprintf("python-%s", p.Name))
+	case pkg.Java:
+		if p.MetadataType == pkg.JavaMetadataType {
+			if metadata, ok := p.Metadata.(pkg.JavaMetadata); ok && metadata.PomProperties != nil {
+				// derive the vendor from the groupID (e.g. org.sonatype.nexus --> sonatype)
+				if strings.HasPrefix(metadata.PomProperties.GroupID, "org.") || strings.HasPrefix(metadata.PomProperties.GroupID, "com.") {
+					fields := strings.Split(metadata.PomProperties.GroupID, ".")
+					if len(fields) >= 3 {
+						vendors = append(vendors, fields[1])
+					}
+				}
+			}
+		}
 	}
 	return vendors
 }
 
 func candidateProducts(p pkg.Package) []string {
-	return []string{p.Name}
+	var products = []string{p.Name}
+	switch p.Language {
+	case pkg.Java:
+		if p.MetadataType == pkg.JavaMetadataType {
```

Comment:
Hey again 😄 Now that I'm seeing this second instance of the same conditional blocks, I think we really should consider factoring this out, and perhaps in a way that elegantly exposes the fields of interest (i.e. vendor and product)